### ISSUE TITLE
Run button doesn't work when workspace from CLI run exists

### ISSUE THREAD

**Issue by emlys** [natcap/invest#322]
Observed on both main and release/3.9
Steps to reproduce:
1. `python setup.py install` and `invest launch` on main
2. Open any model and run it
3. Download and open the invest-mac-binaries artifact from github from the latest build of main
4. Open the InVEST app and open the same model as you ran before
5. Make sure the workspace is the same as the workspace from the previous run. Click the Run button.

Clicking the Run button has no effect when the workspace already exists from a previous run via the CLI. (No log file is created, no window pops up). If you delete the workspace directory, it works. If the previous run was via the app, it works.

The reverse is not true: if the model is launched via the CLI, it doesn't matter whether the workspace was created previously from the CLI or the app.

**Comment by phargogh** [comment]
I confirm that this is also an issue on my mac, though this is also _only_ an issue when running through the launcher as an application.  Reaching into the application itself and running the binaries seems to work just fine:
* `./InVEST.app/Contents/MacOS/InVEST launch` works great
* `./InVEST.app/Contents/MacOS/invest_dist/invest launch` works great

Only launching the application directly does not work.  Examples include:
* double-clicking on `InVEST.app`
* `open ./InVEST.app`

I was testing this on `release/3.9`, revision `7791b7d8c`.

I tried to reproduce this on Windows as well, but could not reproduce the issue.  Running the launcher through the CLI works fine, running the launcher through the binaries does not work.

**Comment by dcdenu4** [comment]
Should this issue be part of the 3.9 milestone?

**Comment by phargogh** [comment]
@dcdenu4 yes, this would prevent a user from running InVEST, so we should absolutely resolve this before the release.  I'll add it to the milestone.

**Comment by emlys** [comment]
Appears related to this console message:
`Sandbox: invest(13237) System Policy: deny(1) file-read-data /Users/emily/Documents/seasonal_water_yield_workspace`

**Comment by emlys** [comment]
Some observations:
* InVEST processes are not sandboxed, according to activity monitor
* Manually giving InVEST full disk access in System preferences > Security and privacy does not fix the problem
* The issue of validation not working the first time is related. There are also several similar messages: `Sandbox: invest(19141) System Policy: deny(1) file-read-xattr /Users/emily/Downloads/DelineateIt/DEM_gura.tif.xml` indicating that the invest app isn't being allowed to read the input files.
* There are even more similar `Sandbox: invest(19161) System Policy: deny(1) file-read-data` messages indicating that the invest app is trying to read the `Preferences` of several unrelated apps. This is happening with these paths:
    * `/Users/emily/Library/Autosave Information`
    * `/Users/emily/Library/Containers/com.apple.VoiceMemos/Data/Library/Preferences`
    * `/Users/emily/Library/Containers/com.apple.archiveutility/Data/Library/Preferences`
    * `/Users/emily/Library/Containers/com.apple.Home/Data/Library/Preferences`
    * `/Users/emily/Library/Containers/com.apple.Safari/Data/Library/Preferences`
    * `/Users/emily/Library/Containers/com.apple.iChat/Data/Library/Preferences`
    * `/Users/emily/Library/Containers/com.apple.CloudDocs.MobileDocumentsFileProvider/Data/Library/Preferences`
    * `/Users/emily/Library/Containers/com.apple.mail/Data/Library/Preferences`
    * `/Users/emily/Library/Containers/com.apple.news/Data/Library/Preferences`
Is this concerning?

**Comment by emlys** [comment]
It sounds like this should be resolved by signing the mac binaries when we get the Stanford apple developer ID certificate.
 
I posted on the Apple Developer forums to get help with this, and got some helpful responses:

    

> 
> > I had thought that TCC would prompt the user for permission if the app
> >      tries to access ~/Documents.
> 
> That is the expected behaviour. I suspect your app has a code signing problem that’s confusing TCC.
> 
>   
> 
> >   I am working on a macOS app which is distributed outside of the App
> >      Store and isn't … signed.
> 
> Yeah, that’s not good. TCC needs a stable signature for it to be able to record which app was granted permission by the user. You need to sign your code. For development, I encourage you to use an Apple Developer signing identity. For distribution, you’ll need Developer ID.

> Pending that you should use an Apple Development signing identity for your day-to-day development. This will give you a stable signature, which should be sufficient to get TCC to behave.
> 
> Two things:
> 
> * It may not actually be sufficient to get TCC to behave (-:  There’s a world of potential issues here, especially when working with third-party libraries and language runtimes.
> 
> * Your Developer ID and Apple Development signing identities looks different to TCC.  If you do your day-to-day debugging with Apple Development, you’ll likely see some TCC oddities when you test your Developer ID app on the same Mac.  This is one of the reasons for the testing procedure I outlined in this post.

**Comment by emlys** [comment]
Update on this: We're now signing the macOS distribution, but this issue is still happening. [This post](https://stackoverflow.com/questions/53749528/my-code-signed-dmg-is-rejected-by-gatekeeper) confirms that apps signed without an Apple certificate still don't pass Gatekeeper, and it seems reasonable that TCC might also require an Apple certificate. We'll revisit this once we get the Apple Developer account.

**Comment by dcdenu4** [comment]
I think it's fair to bump this from the 3.9 milestone as we're waiting on things outside of our control.

**Comment by emlys** [comment]
Closing because won't fix.
